source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -150,23 +152,30 @@ Http::LowerCaseStrPairVector ClientConfig::toHeadersAdd(
   return header_vec;
 }
 
-RawHttpClientImpl::RawHttpClientImpl(Upstream::ClusterManager& cm, ClientConfigSharedPtr config)
-    : cm_(cm), config_(config) {}
+RawHttpClientImpl::RawHttpClientImpl(Upstream::ClusterManager& cm, ClientConfigSharedPtr config,
+                                     TimeSource& time_source)
+    : cm_(cm), config_(config), time_source_(time_source) {}
 
 RawHttpClientImpl::~RawHttpClientImpl() { ASSERT(!callbacks_); }
 
 void RawHttpClientImpl::cancel() {
   ASSERT(callbacks_ != nullptr);
+  span_->setTag(Tracing::Tags::get().Status, Tracing::Tags::get().Canceled);
+  span_->finishSpan();
   request_->cancel();
   callbacks_ = nullptr;
```

Comment:
Should we `span_ = nullptr` as well and assert that it's null in the destructor?